<table align="center" width=100%>
    <tr>
        <td width="15%">
            <img src="edaicon.png">
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=24px>
                    <b>Exploratory Data Analysis
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

## Problem Statement

The zomato exploratory data analysis is for the foodies to find the best restaurants, value for money restaurants in their locality. It also helps to find their required cuisines in their locality.  

## Data Definition

**res_id**: The code given to a restaurant (Categorical) 

**name**: Name of the restaurant (Categorical)

**establishment**: Represents the type of establishment (Categorical)

**url**: The website of the restaurant (Categorical)

**address**: The address of the restaurant (Categorical)

**city**: City in which the restaurant located (Categorical)

**city_id**: The code given to a city (Categorical)

**locality**: Locality of the restaurant (Categorical)

**latitude**: Latitude of the restaurant (Categorical)

**longitude**: Longitude of the restaurant (Categorical)

**zipcode**: Zipcode of the city in which the restaurant located (Categorical)

**country_id**: Country code in which the restaurant located (Categorical)

**locality_verbose**: Locality along with the city in which the restaurant located (Categorical)

**cuisines**: The cuisines a restaurant serves (Categorical)

**timings**: The working hours of a restaurant (Categorical)

**average_cost_for_two**: The average amount expected for 2 people (Numerical)

**price_range**: The categories for average cost (Categories - 1,2,3,4) (Categorical)

**currency**: The currency in which a customer pays (Categorical)

**highlights**: The facilities of the restaurant (Categorical)

**aggregate_rating**: The overall rating a restaurant has got (Numerical) 

**rating_text**: Categorized ratings (Categorical)

**votes**: Number of votes received by the restaurant from customers (Numerical)

**photo_count**: The number of photos of a restaurant (Numerical)

**opentable_support**: Restaurant reservation from Opentable (Categorical)

**delivery**: The restaurant deliver an order or not (Categorical)

**takeaway**: The restaurant allows  a 'takeaway' of an order or not (Categorical)

## Table of Contents

1. **[Import Libraries](#import_lib)** 
2. **[Set Options](#set_options)** 
3. **[Read Data](#Read_Data)** 
4. **[Understand  and Prepare the Data](#Understand_Data)**
5. **[Understand the variables](#Understanding_variables)**
6. **[Check for Missing Values](#missing)**
7. **[Study Correlation](#correlation)**
8. **[Detect Outliers](#outliers)**
9. **[Create a new variable 'region'](#region)**
10. **[Some more analysis](#more)** 


<a id='import_lib'></a>
## 1. Import Libraries

<table align ="left">
    <tr>
        <td width="8%">
            <img src="todo.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b> Import the required libraries and functions
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

In [1]:
import pandas as pd
import numpy as np 

<a id='set_options'></a>
## 2. Set Options

<table align="left">
    <tr>
        <td width="8%">
            <img src="todo.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>Make necessary changes to :<br><br>
Set the working directory              
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

<a id='Read_Data'></a>
## 3. Read Data

In [2]:
data=pd.read_csv("ZomatoRestaurantsIndia.csv")
data.head()

res_id                            name  establishment  \
0  3400299                     Bikanervala    Quick Bites   
1  3400005  Mama Chicken Mama Franky House    Quick Bites   
2  3401013                   Bhagat Halwai    Quick Bites   
3  3400290                   Bhagat Halwai    Quick Bites   
4  3401744     The Salt Cafe Kitchen & Bar  Casual Dining   

                                                 url  \
0  https://www.zomato.com/agra/bikanervala-khanda...   
1  https://www.zomato.com/agra/mama-chicken-mama-...   
2  https://www.zomato.com/agra/bhagat-halwai-2-sh...   
3  https://www.zomato.com/agra/bhagat-halwai-civi...   
4  https://www.zomato.com/agra/the-salt-cafe-kitc...   

                                             address  city  city_id  \
0  Kalyani Point, Near Tulsi Cinema, Bypass Road,...  Agra       34   
1        Main Market, Sadar Bazaar, Agra Cantt, Agra  Agra       34   
2  62/1, Near Easy Day, West Shivaji Nagar, Goalp...  Agra       34   
3  Near Anjana Cinema, Nehru Nagar, Civil Lines, ...  Agra       34   
4        1C,3rd Floor, Fatehabad Road, Tajganj, Agra  Agra       34   

      locality   latitude  longitude  ... price_range  currency  \
0     Khandari  27.211450  78.002381  ...           2       Rs.   
1   Agra Cantt  27.160569  78.011583  ...           2       Rs.   
2     Shahganj  27.182938  77.979684  ...           1       Rs.   
3  Civil Lines  27.205668  78.004799  ...           1       Rs.   
4      Tajganj  27.157709  78.052421  ...           3       Rs.   

                                          highlights aggregate_rating  \
0  Lunch, Takeaway Available, Credit Card, Dinner...              4.4   
1  Delivery, No Alcohol Available, Dinner, Takeaw...              4.4   
2  No Alcohol Available, Dinner, Takeaway Availab...              4.2   
3  Takeaway Available, Credit Card, Lunch, Delive...              4.3   
4  Lunch, Serves Alcohol, Cash, Credit Card, Dinn...              4.9   

  rating_text  votes  photo_count opentable_support delivery  takeaway  
0   Very Good    814          154               0.0       -1        -1  
1   Very Good   1203          161               0.0       -1        -1  
2   Very Good    801          107               0.0        1        -1  
3   Very Good    693          157               0.0        1        -1  
4   Excellent    470          291               0.0        1        -1  

[5 rows x 26 columns]

<a id='Understand_Data'></a>
## 4. Understand  and Prepare the Data

A well-prepared data proves beneficial for analysis as it limits errors and inaccuracies that can occur during analysis. The processed data is more accessible to users.<br> <br>
                         Data understanding is the process of getting familiar with the data, to identify data type, to discover first insights into the data, or to detect interesting subsets to form hypotheses about hidden information. Whereas, data preparation is the process of cleaning and transforming raw data before analysis. It is an important step before processing and often involves reformatting data, making corrections to data. <br> <br>
                        Data preparation is often a lengthy process, but it is essential as a prerequisite to put data in context to get insights and eliminate bias resulting from poor data quality.

<table align="left">
    <tr>
        <td width="8%">
            <img src="todo.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b> Analyze and prepare data:<br>
                        1. Check dimensions of the dataframe <br>
                        2. View the head of the data<br>
                        3. Note the redundant variables and drop them <br>
                        4. Check the data types. Refer to data definition to ensure your data types are correct. If data types are not as per business context, change the data types as per requirement <br>
                        5. Check for duplicates<br>
                        Note: It is an art to explore data and one will need more and more practice to gain expertise in this area
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

### -------------------------*** Provide the inference's from the output of every code executed.***----------------------------

**1. Check dimensions of the dataframe in terms of rows and columns**

In [3]:
data.shape

(211944, 26)

Inference drawn: 
    * The no.of rows are 211944
    * The no.of columns are 26

**2. View the head of the data**

In [4]:
data.head(5)
# data.locality_verbose

res_id                            name  establishment  \
0  3400299                     Bikanervala    Quick Bites   
1  3400005  Mama Chicken Mama Franky House    Quick Bites   
2  3401013                   Bhagat Halwai    Quick Bites   
3  3400290                   Bhagat Halwai    Quick Bites   
4  3401744     The Salt Cafe Kitchen & Bar  Casual Dining   

                                                 url  \
0  https://www.zomato.com/agra/bikanervala-khanda...   
1  https://www.zomato.com/agra/mama-chicken-mama-...   
2  https://www.zomato.com/agra/bhagat-halwai-2-sh...   
3  https://www.zomato.com/agra/bhagat-halwai-civi...   
4  https://www.zomato.com/agra/the-salt-cafe-kitc...   

                                             address  city  city_id  \
0  Kalyani Point, Near Tulsi Cinema, Bypass Road,...  Agra       34   
1        Main Market, Sadar Bazaar, Agra Cantt, Agra  Agra       34   
2  62/1, Near Easy Day, West Shivaji Nagar, Goalp...  Agra       34   
3  Near Anjana Cinema, Nehru Nagar, Civil Lines, ...  Agra       34   
4        1C,3rd Floor, Fatehabad Road, Tajganj, Agra  Agra       34   

      locality   latitude  longitude  ... price_range  currency  \
0     Khandari  27.211450  78.002381  ...           2       Rs.   
1   Agra Cantt  27.160569  78.011583  ...           2       Rs.   
2     Shahganj  27.182938  77.979684  ...           1       Rs.   
3  Civil Lines  27.205668  78.004799  ...           1       Rs.   
4      Tajganj  27.157709  78.052421  ...           3       Rs.   

                                          highlights aggregate_rating  \
0  Lunch, Takeaway Available, Credit Card, Dinner...              4.4   
1  Delivery, No Alcohol Available, Dinner, Takeaw...              4.4   
2  No Alcohol Available, Dinner, Takeaway Availab...              4.2   
3  Takeaway Available, Credit Card, Lunch, Delive...              4.3   
4  Lunch, Serves Alcohol, Cash, Credit Card, Dinn...              4.9   

  rating_text  votes  photo_count opentable_support delivery  takeaway  
0   Very Good    814          154               0.0       -1        -1  
1   Very Good   1203          161               0.0       -1        -1  
2   Very Good    801          107               0.0        1        -1  
3   Very Good    693          157               0.0        1        -1  
4   Excellent    470          291               0.0        1        -1  

[5 rows x 26 columns]

**3. Note the redundant variables and drop them**

In [5]:
data.columns

#country_id
#city
#country_id
#locality_verbose
#currency
#opentable_support

data=data.drop(['country_id','locality_verbose','currency','opentable_support'],axis=1)


In [6]:
data.shape

(211944, 22)

**4. Check the data types. Refer to data definition to ensure your data types are correct. If data types are not as per business context, change the data types as per requirement**


In [7]:
# data.astype()


#### Change the incorrect data type

**5. Check for Duplicates**

<a id = 'Understanding_variables'> </a>
## 5. Understand the variables

**1. Variable 'name'**

In [8]:
data.name

0                            Bikanervala
1         Mama Chicken Mama Franky House
2                          Bhagat Halwai
3                          Bhagat Halwai
4            The Salt Cafe Kitchen & Bar
                       ...              
211939    Kali Mirch Cafe And Restaurant
211940                        Raju Omlet
211941                  The Grand Thakar
211942                            Subway
211943        Freshcos - The Health Cafe
Name: name, Length: 211944, dtype: object

**2. Variable 'establishment'**

In [9]:
data.establishment

0           Quick Bites
1           Quick Bites
2           Quick Bites
3           Quick Bites
4         Casual Dining
              ...      
211939    Casual Dining
211940      Quick Bites
211941    Casual Dining
211942      Quick Bites
211943             Café
Name: establishment, Length: 211944, dtype: object

**3. Variable 'city'**

In [10]:
data.city

0             Agra
1             Agra
2             Agra
3             Agra
4             Agra
            ...   
211939    Vadodara
211940    Vadodara
211941    Vadodara
211942    Vadodara
211943    Vadodara
Name: city, Length: 211944, dtype: object

**Let us find the count of restaurants in each city**

In [11]:
data.city.value_counts()


Chennai          11630
Mumbai            6497
Bangalore         4971
Pune              4217
Lucknow           4121
                 ...  
Ghaziabad          132
Faridabad          124
Howrah              66
Greater Noida       33
Nayagaon            17
Name: city, Length: 98, dtype: int64

**4. Variable 'locality'**

In [12]:
data.locality

0            Khandari
1          Agra Cantt
2            Shahganj
3         Civil Lines
4             Tajganj
             ...     
211939      Fatehgunj
211940     Karelibaug
211941       Alkapuri
211942          Akota
211943       Vadiwadi
Name: locality, Length: 211944, dtype: object

**4. Variable 'latitude'**

From the variable 'latitude', we know the latitudinal location of the restaurant

The Latitudinal extent of India 8º4‛N to 37º6‛ N. 

We must check whether we have any points beyond this extent.

- We need to replace all these values with NaN's.

- check if the values are replace by NaN's

- We see all the values are replaced by NaN's

In [13]:
data.latitude

0         27.211450
1         27.160569
2         27.182938
3         27.205668
4         27.157709
            ...    
211939    22.336931
211940    22.322455
211941    22.310563
211942    22.270027
211943    22.309935
Name: latitude, Length: 211944, dtype: float64

In [14]:
data.latitude.isna().sum()

0

In [15]:
data.loc[(data.latitude<8.4)| (data.latitude >37.6),['latitude']]=None

In [16]:
data.latitude.isna().sum()

1339

**5. Variable 'longitude'**

From the variable 'longitude', we know the longitudinal location of the restaurant

The Longitudinal extent of India is from 68°7'E to 97°25'E

We must check whether we have any points beyond this extent.

- We need to replace all these values with NaN's.

- Check if the values are replace by NaN's

- From variable 'latitude' and 'longitude', plot the location of restaurants.

In [17]:
data.longitude

0         78.002381
1         78.011583
2         77.979684
3         78.004799
4         78.052421
            ...    
211939    73.192356
211940    73.197203
211941    73.171164
211942    73.143068
211943    73.158768
Name: longitude, Length: 211944, dtype: float64

In [18]:
data.longitude.isna().sum()

0

In [19]:
data.loc[(data.longitude<68.7)| (data.longitude >97.25),['longitude']]=None

In [20]:
data.longitude.isna().sum()

1329

**6. Variable 'cuisines'**

In [21]:
data.cuisines

0         North Indian, South Indian, Mithai, Street Foo...
1         North Indian, Mughlai, Rolls, Chinese, Fast Fo...
2                                         Fast Food, Mithai
3                 Desserts, Bakery, Fast Food, South Indian
4                        North Indian, Continental, Italian
                                ...                        
211939                                         North Indian
211940                                            Fast Food
211941                      Gujarati, North Indian, Chinese
211942                           Fast Food, Sandwich, Salad
211943                           Cafe, Healthy Food, Coffee
Name: cuisines, Length: 211944, dtype: object

- To find the unique cusines we write a small user defined function.



In [22]:
save_cuisines=(list(map(str,(data.cuisines))))
result_cuisines=set()
for i in save_cuisines:
    inter=(i.split(", "))
    for j in inter:
        result_cuisines.add(j)


print(result_cuisines)



{'Lebanese', 'Beverages', 'Roast Chicken', 'Bangladeshi', 'Pakistani', 'Belgian', 'Juices', 'Continental', 'Maharashtrian', 'Burger', 'Korean', 'German', 'Israeli', 'Chettinad', 'Iranian', 'Cuisine Varies', 'Mexican', 'Pan Asian', 'Sindhi', 'North Eastern', 'Arabian', 'Biryani', 'Bengali', 'Naga', 'Street Food', 'European', 'Kerala', 'Hyderabadi', 'Paan', 'Awadhi', 'Armenian', 'Hot dogs', 'American', 'Assamese', 'Brazilian', 'Indonesian', 'Afghani', 'Portuguese', 'Spanish', 'Andhra', 'Tibetan', 'Mishti', 'Middle Eastern', 'Bohri', 'Italian', 'Vietnamese', 'Coffee and Tea', 'Cafe Food', 'Frozen Yogurt', 'Himachali', 'Swedish', 'Cantonese', 'Mughlai', 'Malwani', 'Sri Lankan', 'International', 'Steak', 'Modern Australian', 'Grill', 'Mithai', 'Bihari', 'Burmese', 'nan', 'French', 'Tex-Mex', 'Turkish', 'Thai', 'Mongolian', 'Goan', 'Healthy Food', 'Gujarati', 'Bakery', 'Coffee', 'Malaysian', 'Mediterranean', 'Singaporean', 'Oriental', 'Afghan', 'Mangalorean', 'Bar Food', 'Irish', 'Momos', 'R

- find out the frequency of each cuisine

In [23]:
result_cuisines=list(result_cuisines)

data.cuisines.value_counts()

North Indian                                                         15996
Fast Food                                                             6721
Cafe                                                                  6190
North Indian, Chinese                                                 5820
South Indian                                                          5217
                                                                     ...  
Asian, North Indian, American, Street Food, Healthy Food, Italian        1
BBQ, Middle Eastern, Moroccan, Hyderabadi, Lebanese, Mughlai             1
South Indian, North Indian, Juices, Chinese                              1
Bakery, Fast Food, Pizza, Burger, Chinese, Sandwich, Wraps, Rolls        1
Fast Food, Tea, Desserts                                                 1
Name: cuisines, Length: 9382, dtype: int64

**8. Variable 'average_cost_for_two'**

In [24]:
data.average_cost_for_two

0          700
1          600
2          300
3          300
4         1000
          ... 
211939     600
211940     300
211941     700
211942     500
211943     600
Name: average_cost_for_two, Length: 211944, dtype: int64

**9. Variable 'price_range'**

In [ ]:
data.price_range.plot.pie(figsize=(6,6))

- visualize a exploded pie chart.

**10. Variable 'highlights'**

In [35]:
data.highlights

0         Lunch, Takeaway Available, Credit Card, Dinner...
1         Delivery, No Alcohol Available, Dinner, Takeaw...
2         No Alcohol Available, Dinner, Takeaway Availab...
3         Takeaway Available, Credit Card, Lunch, Delive...
4         Lunch, Serves Alcohol, Cash, Credit Card, Dinn...
                                ...                        
211939    Dinner, Cash, Lunch, Delivery, Indoor Seating,...
211940    Dinner, Cash, Takeaway Available, Debit Card, ...
211941    Dinner, Cash, Debit Card, Lunch, Takeaway Avai...
211942    Dinner, Delivery, Credit Card, Lunch, Cash, Ta...
211943    Dinner, Cash, Takeaway Available, Debit Card, ...
Name: highlights, Length: 211944, dtype: object



- write a small function to know the number of times a facility has appeared in the 'Highlights'.

- Now we find out which facility occurs most number of in the data.

**11. Variable 'aggregate_rating'** 



In [26]:
data.aggregate_rating

0         4.4
1         4.4
2         4.2
3         4.3
4         4.9
         ... 
211939    4.1
211940    4.1
211941    4.0
211942    3.7
211943    4.0
Name: aggregate_rating, Length: 211944, dtype: float64

**12. Variable 'rating_text'**



In [27]:
data.rating_text

0         Very Good
1         Very Good
2         Very Good
3         Very Good
4         Excellent
            ...    
211939    Very Good
211940    Very Good
211941    Very Good
211942         Good
211943    Very Good
Name: rating_text, Length: 211944, dtype: object

Creating a New feature for better understanding of ratings


In [32]:
data.rating_text.unique()

array(['Very Good', 'Excellent', 'Good', 'Average', 'Not rated', 'Poor',
       'Dobré', 'Baik', 'Sangat Baik', 'Excelente', 'Bardzo dobrze',
       'Wybitnie', 'Ottimo', 'Muito Bom', 'Velmi dobré', 'Skvělá volba',
       'Muy Bueno', 'Bom', 'İyi', 'Çok iyi', 'Harika', 'Terbaik',
       'Skvělé', 'Průměr', 'Ortalama', 'Bueno', 'Eccellente', 'Muito bom',
       'Dobrze', 'Buono', 'Média', 'Scarso', 'Promedio', 'Veľmi dobré',
       'Vynikajúce', 'Średnio', 'Priemer', 'Biasa', 'Media'], dtype=object)

**13. Variable 'votes'**

In [33]:
data.votes

0          814
1         1203
2          801
3          693
4          470
          ... 
211939     243
211940     187
211941     111
211942     128
211943      93
Name: votes, Length: 211944, dtype: int64

**14. Variable 'photo_count'**

In [34]:
data.photo_count

0         154
1         161
2         107
3         157
4         291
         ... 
211939     40
211940     40
211941     38
211942     34
211943     53
Name: photo_count, Length: 211944, dtype: int64

**15. Variable 'delivery'**

In [31]:
data.delivery

0        -1
1        -1
2         1
3         1
4         1
         ..
211939   -1
211940    1
211941   -1
211942    1
211943    1
Name: delivery, Length: 211944, dtype: int64

<a id ='missing'></a>
## 6. Check for missing values

In [37]:
data.isna().sum()

res_id                       0
name                         0
establishment             4827
url                          0
address                    134
city                         0
city_id                      0
locality                     0
latitude                  1339
longitude                 1329
zipcode                 163187
cuisines                  1391
timings                   3874
average_cost_for_two         0
price_range                  0
highlights                2069
aggregate_rating             0
rating_text                  0
votes                        0
photo_count                  0
delivery                     0
takeaway                     0
dtype: int64

**6. Study summary statistics**

Let us check the summary statistics for numerical variables.

In [38]:
data.describe()

res_id        city_id       latitude      longitude  \
count  2.119440e+05  211944.000000  210605.000000  210615.000000   
mean   1.349411e+07    4746.785434      21.586832      78.104827   
std    7.883722e+06    5568.766386       6.823595       4.286852   
min    5.000000e+01       1.000000       8.400144      69.014446   
25%    3.301027e+06      11.000000      15.545368      74.892431   
50%    1.869573e+07      34.000000      22.541707      77.434536   
75%    1.881297e+07   11306.000000      26.847055      80.221799   
max    1.915979e+07   11354.000000      34.214702      91.832769   

       average_cost_for_two    price_range  aggregate_rating          votes  \
count         211944.000000  211944.000000     211944.000000  211944.000000   
mean             595.812229       1.882535          3.395937     378.002571   
std              606.239363       0.892989          1.283642     925.333106   
min                0.000000       1.000000          0.000000       0.000000   
25%              250.000000       1.000000          3.300000      16.000000   
50%              400.000000       2.000000          3.800000     100.000000   
75%              700.000000       2.000000          4.100000     362.000000   
max            30000.000000       4.000000          4.900000   42539.000000   

         photo_count       delivery  takeaway  
count  211944.000000  211944.000000  211944.0  
mean      256.971224      -0.255907      -1.0  
std       867.668940       0.964172       0.0  
min         0.000000      -1.000000      -1.0  
25%         3.000000      -1.000000      -1.0  
50%        18.000000      -1.000000      -1.0  
75%       128.000000       1.000000      -1.0  
max     17702.000000       1.000000      -1.0

<a id = 'correlation'> </a>
## 7. Study correlation

In [ ]:
import seaborn as sns
corr=data.corr()
sns.heatmap(corr,annot=True)



## 8. Detect outliers

<a id='region'> </a>
## 9. Create a new variable 'region'


Create a variable 'region' with four categories 'northern','eastern', 'southern', 'western' and 'central'. To do so, use the 'city' column, group all cities belonging to the same region. 

<a id='more'> </a>
## 10. Some more Analysis

<b>Lets us explore the data some more now that we have extrapolated and removed the missing values <br>
We now conduct analysis to compare the regions.</b>

### 1. To find which cities have expensive restaurants 

-  plot the cities which have costliest restaurants. 

### 2.  Comparing regions

### 2a. Highlights available in restaurants for different regions

To cater our analysis we define the regions as nothern, eastern, western and southern.

We first need to select the unique facilities available in each region and sort according to their frequencies.

**Highlights of the northern region**

**Highlights of the eastern region**

**Highlights of the southern region**

**Highlights of the western region**

#### Plot the barplot for different regions

We shall now plot the graphs for top 10 highlights.

### 2b. Cuisines available in restaurants for different regions

**Cuisines in the northern region**

**Cuisines in the eastern region**

**Cuisines in the southern region**

**Cuisines in the western region** 

- Plot the barplot for top 10 cuisines served in the four regions

###  3. The Northern Region

**Now we shall consider only the northern region**

**1. The top 10 cuisines served in Restaurants** 

**2. Do restaurants with more photo counts and votes have better rating?**

- Plot a boxplots for the above table

### 4. The Mumbai city

consider the city mumbai and get a better insights of restuarants in Mumbai.

**1. Expensive restaurants in Mumbai**

-  Define the costliest restaurants whose average cost of two people exceeds Rs.5000 .
-  Plot the restaurants which are costliest based on their average cost for two .




**2.To find the top 20 cuisines of Mumbai**

- select unique cuisines available at restaurants in Mumbai


- sort cuisines based on frequency


**3. To find the popular localities in Mumbai**

**4. Check for relationship between 'aggregate_rating' and 'average_cost_for_two'**

**5. Multiple box plot for photo_counts based on establishment type.**



**6. Check for payments method offered in restaurants**

- select unique facilities available at restaurants in western region
- sort facilities based on frequency
